File: frontend/src/types/linked-account.ts
export type OAuthProvider = "google" | "github" | "facebook"

export interface LinkedAccount {
  id: string
  provider: OAuthProvider
  provider_user_id: string
  provider_email: string
  created_at: string
}

export interface LinkedAccountsResponse {
  linked_accounts: LinkedAccount[]
  available_providers: string[]
}

export interface UnlinkAccountRequest {
  provider: OAuthProvider
  password?: string
}

export interface UnlinkAccountResponse {
  message: string
  remaining_providers: string[]
}

File: backend/app/schemas/linked_account.py
"""Linked Account schemas for OAuth account linking."""

from datetime import datetime
from uuid import UUID
from pydantic import BaseModel, Field

from app.models import OAuthProvider


class LinkedAccountBase(BaseModel):
    provider: OAuthProvider
    provider_email: str


class LinkedAccountPublic(LinkedAccountBase):
    """Public representation of a linked account"""
    id: UUID
    provider_user_id: str
    created_at: datetime

    class Config:
        from_attributes = True


class LinkedAccountsResponse(BaseModel):
    """Response containing all linked accounts for a user"""
    linked_accounts: list[LinkedAccountPublic]
    available_providers: list[str]  # Providers not yet linked


class LinkAccountRequest(BaseModel):
    """Request to initiate account linking"""
    provider: OAuthProvider


class UnlinkAccountRequest(BaseModel):
    """Request to unlink an account"""
    provider: OAuthProvider
    password: str | None = Field(
        default=None,
        description="Required if this is the last login method and user has no password"
    )


class UnlinkAccountResponse(BaseModel):
    """Response after unlinking an account"""
    message: str
    remaining_providers: list[str]


class LinkCallbackResponse(BaseModel):
    """Response after successfully linking an account"""
    message: str
    provider: str
    provider_email: str
